extensions/markdown-language-features/src/preview/preview.ts

```diff
@@ -110,6 +110,7 @@ class MarkdownPreview extends Disposable implements WebviewResourceProvider {
 			}
 		}));
 
+		// Watch for changes to files in the same directory (for references, etc.)
 		const watcher = this._register(vscode.workspace.createFileSystemWatcher(new vscode.RelativePattern(resource, '*')));
```

Comment:
We should only need one watcher. Also have you debugged why this doesn't work? It seems to work fine in my testing